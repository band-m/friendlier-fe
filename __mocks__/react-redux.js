jest.mock('react-redux', () => {
  return {
    __esModule: true,
    useDispatch: jest.fn(() => () => {}),
    useSelector: jest.fn((selector) => selector({ contacts: { contactList: [] }, auth: { user: '' } }))
  };
});
