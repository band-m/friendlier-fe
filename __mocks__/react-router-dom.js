jest.mock('react-router-dom', () => {
  return {
    useHistory: jest.fn({})
  };
});
