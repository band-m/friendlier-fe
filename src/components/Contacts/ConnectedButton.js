import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch, useSelector } from 'react-redux';
import { myAction, editContactDetails } from '../../data/actions/contact-detail-actions';
import { SET_LAST_CONTACTED_DATE, SET_YELLOW_ZONE, SET_RED_ZONE, SET_DEADLINE_DATE, SET_CONTACT_DETAILS } from '../../data/action-types/action-types';
import add from 'date-fns/add';
import { useHistory } from 'react-router-dom';
import { selectContactDetails } from '../../data/selectors/contact-detail-selectors';

export default function ConnectedButton({ slider1, slider2, deadlineObject, id }){
  const contact = useSelector(selectContactDetails);
  
  const history = useHistory();
  
  const dispatch = useDispatch();

  const addConnectionEvent = () => {
    const updatedContact = {
      ...contact,
      yellowZoneStartDate: add(new Date(), { days: slider1 }),
      redZoneStartDate: add(new Date(), { days: slider2 }),
      deadlineDate: add(new Date(), deadlineObject),
      lastContactedDate: new Date(),
      connHistory: [ ...contact.connHistory, new Date() ]
    }
    dispatch(editContactDetails(id, updatedContact));
    history.push('/contacts');
  }

  return (
    <button onClick={() => addConnectionEvent()}>Add connection event!</button>
  );
};

ConnectedButton.propTypes = {
  slider1: PropTypes.number.isRequired,
  slider2: PropTypes.number.isRequired,
  deadlineObject: PropTypes.object.isRequired
};
