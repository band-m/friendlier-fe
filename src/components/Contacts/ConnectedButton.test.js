import React from 'react';
import { shallow } from 'enzyme';
import 'react-redux';
import 'react-router-dom';
import ConnectedButton from './ConnectedButton';

describe('ConnectedButton component', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<ConnectedButton slider1={''} slider2={''} deadlineObject={{}} id={''} />);
    expect(wrapper).toMatchSnapshot();
  });
});
