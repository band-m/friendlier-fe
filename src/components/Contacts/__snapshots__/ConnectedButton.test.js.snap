// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedButton component matches snapshot 1`] = `
<button
  onClick={[Function]}
>
  Add connection event!
</button>
`;
